src/services/item/plugins/published/plugins/search/index.ts

```diff
@@ -17,12 +19,21 @@ export type SearchFields = {
 
 const plugin: FastifyPluginAsync = async (fastify) => {
   const searchService = fastify.search.service;
+  const actionService = fastify.actions.service;
 
-  fastify.post(
+  fastify.post<{ Body: MultiSearchParams }>(
     '/collections/search',
     { preHandler: fastify.attemptVerifyAuthentication, schema: search },
-    async ({ member, body }) => {
-      return searchService.search(member, buildRepositories(), body as MultiSearchParams);
+    async (request) => {
+      const { member, body } = request;
+      const repositories = buildRepositories();
+
+      const action = {
+        type: Triggers.ItemSearch,
+        extra: body,
+      };
+      await actionService.postMany(member, repositories, request, [action]);
+      return searchService.search(member, repositories, body);
```

Comment:
Maybe let's perform the search first, then the post action. Otherwise you might end up having an action even though the search failed.